[jit] Break interpreter.cpp into smaller files.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57398 WIP _ScriptProfile
* #57397 [jit] Implement ScriptProfile to collect instruction profiles.
* #57396 [jit] Add a utility class SourceRef to represent Source as keys
* **#56546 [jit] Break interpreter.cpp into smaller files.**

Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133580](https://our.internmc.facebook.com/intern/diff/D28133580)

zhxchen17 committed:
WIP

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zhxchen17 committed:
Update on "WIP"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Break interpreter.cpp into smaller files."


Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

gmagogsfm commented on torch/csrc/jit/runtime/interpreter.cpp:
```diff
@@ -431,667 +103,6 @@ struct TLSCurrentInterpreterGuard {
-  }
-};
-
 struct MobileCodeImpl : CodeImpl {
```
Could we move `MobileCodeImpl` to `code_impl.{h, cpp}` as well?

zhxchen17 replied:
sure.

gmagogsfm reviewed (approved):
Looks good (assuming there are no functional changes), thanks!

zhxchen17 committed:
addressed suggestions on "[jit] Break interpreter.cpp into smaller files."


Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133580](https://our.internmc.facebook.com/intern/diff/D28133580)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).

zhxchen17 committed:
nits on "[jit] Break interpreter.cpp into smaller files."


Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133580](https://our.internmc.facebook.com/intern/diff/D28133580)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).

zhxchen17 committed:
rebase on "[jit] Break interpreter.cpp into smaller files."


Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133580](https://our.internmc.facebook.com/intern/diff/D28133580)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).
